### ISSUE TITLE
Fix S4830 FN: Match DangerousAcceptAnyServerCertificateValidator

### ISSUE THREAD

**Issue by kevinoid** [SonarSource/sonar-dotnet#3824]
### Description

[`HttpClientHandler.DangerousAcceptAnyServerCertificateValidator`](https://docs.microsoft.com/dotnet/api/system.net.http.httpclienthandler.dangerousacceptanyservercertificatevalidator) was added in dotnet/corefx#19908 (.NET Core 2.0) to have the same behavior as a delegate returning true, including on backends that do not support callbacks.  S4830 does not trigger when this property is used instead of a delegate, even though they have the same effect.

### Repro steps

```csharp
using System.Net.Http;

namespace NoServerCertValidation
{
    public static class NoServerCertValidationTest
    {
	public static void WithHandlerDelegate()
	{
	    new HttpClientHandler().ServerCertificateCustomValidationCallback =
		(sender, certificate, chain, sslErrors) => true;
	}

	public static void WithHandlerProp()
	{
	    new HttpClientHandler().ServerCertificateCustomValidationCallback =
		HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
	}
    }
}
```

### Expected behavior

S4830 on both line 9 and 15.

### Actual behavior

S4830 on line 9 only.

### Related information

Using [SonarAnalyzer.CSharp](https://www.nuget.org/packages/SonarAnalyzer.CSharp/) version 8.15.0.24505 building with .NET 5.0:

<details>
<summary><code>dotnet --info</code></summary>

    .NET SDK (reflecting any global.json):
     Version:   5.0.101
     Commit:    d05174dc5a
    
    Runtime Environment:
     OS Name:     debian
     OS Version:  
     OS Platform: Linux
     RID:         debian-x64
     Base Path:   /usr/share/dotnet/sdk/5.0.101/
    
    Host (useful for support):
      Version: 5.0.1
      Commit:  b02e13abab
    
    .NET SDKs installed:
      5.0.101 [/usr/share/dotnet/sdk]
    
    .NET runtimes installed:
      Microsoft.AspNetCore.App 5.0.1 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
      Microsoft.NETCore.App 5.0.1 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
    
    To install additional .NET runtimes or SDKs:
      https://aka.ms/dotnet-download

</details>

**Comment by pavel-mikula-sonarsource** [comment]
Hi @kevinoid, that's a good catch! Thank you for reporting this case.

**Comment by kevinoid** [comment]
Looks great.  Thanks @pavel-mikula-sonarsource!
